### PR TITLE
ql-qlf: k6n10f: update TDP36K simulation models

### PR DESCRIPTION
**Pull request by lpawelcz**
This PR updates TDP BRAM simulation models for k6n10f devices. The update is fixing the issue reported by @rakeshm75 in https://github.com/SymbiFlow/yosys-f4pga-plugins/pull/235#issuecomment-1061380585. The PR also adds a simulation test for 36bit FIFO variant that checks consistency of data storage and occurrences of `EMPTY` and `FULL` flags.
Getting verilog simulation tests to work however, required removing `transport delay` constructs from the whole `fifo_ctl` module.

### CODE REVIEW

**Comment by lpawelcz** [comment]
@rakeshm75 Please have a look at this PR.

**Comment by lpawelcz** [comment]
@tpagarani, @rakeshm75, please take a look at this update. Is this okay to be merged?

**Comment by rakeshm75** [comment]
Here are some observations and required modifications:

**In ufifo_ctl.v file:**

  On line 169 change:

    paf_thresh = (wmode[1] ? (wmode[0] ? upaf << 1 : upaf) : upaf << 2);

  to:

    paf_thresh = wmode[1] ? upaf : (wmode[0] ? upaf << 1 : upaf << 2) ;

  On line 488 change:

    pae_thresh = (rmode ? (rmode[0] ? upae << 1 : upae) : upae << 2);

  to:

    pae_thresh = rmode[1] ? upae : (rmode[0] ? upae << 1 : upae << 2);

**Comment by rakeshm75** [comment]
Sure, going forward I shall provide the feedback as mentioned above. There are more changes required in the BRAM and DSP models.

In our call, Maciej mentioned that you people are referring to the factor repo for the BRAM and DSP IPs. The BRAM and DSP IPs are not up to date on the factor repo. The updated BRAM and DSP IPs are present in the australis repo:

               https://github.com/QuickLogic-Corp/australis

Please sync up the DSP and BRAM sim models with the updated IPs in the australis repo:

         https://github.com/QuickLogic-Corp/australis/tree/main/IPs/eFPGA/IP

**Comment by rakeshm75** [comment]
For the simulation model, please use the sram1024x18.v file itself... functionally its same as the actual ram used.

**Comment by lpawelcz** [comment]
@rakeshm75 Thank you for the answer. I've updated the simulation modules to match the state from the australis repository. Please have a look at this and let me know if this can be merged.
I also updated PR that introduces `split mode` for `TDP36K` module (https://github.com/chipsalliance/yosys-f4pga-plugins/pull/286).

**Comment by rakeshm75** [comment]
I have reviewed the changes and it looks fine. One observation is that I see scan chain signals are added in the simulation model. It would be good if we can remove the scan chain signals and the parallel load signals (PL_*_*) from the simulation model.

**Comment by lpawelcz** [comment]
OK, signals `PL_*` and `SCAN_*` are now removed from both simulation models. Techmaps and tests are also updated accordingly.

**Comment by lpawelcz** [comment]
@rakeshm75 `pushflags` assignment in `ufifo_ctl.v` is now fixed. Please let me know if this PR is ready for merge.

**Comment by rakeshm75** [comment]
Its fine to merge now.

File: ql-qlf-plugin/qlf_k6n10f/brams_map.v [hunk 1]
```diff
@@ -607,23 +547,7 @@ module \$__QLF_FACTOR_BRAM36_SDP (CLK2, CLK3, A1ADDR, A1DATA, A1EN, B1ADDR, B1DA
...
-		.PL_DATA_i(PL_DATA_i),
-		.PL_INIT_o(PL_INIT_o),
-		.PL_ENA_o(PL_ENA_o),
-		.PL_WEN_o(PL_WEN_o),
-		.PL_REN_o(PL_REN_o),
-		.PL_CLK_o(PL_CLK_o),
-		.PL_ADDR_o(),
-		.PL_DATA_o(PL_DATA_o)
-
+		.RAM_ID_i(RAM_ID)
```
[the remark this answers is not in the document]
Comment (lpawelcz) [reply ql-qlf-plugin/qlf_k6n10f/brams_map.v]: `RAM_ID` is now removed form the techmaps and simulation models

[the remark this answers is not in the document]
Comment (tpagarani) [reply ql-qlf-plugin/qlf_k6n10f/brams_map.v]: Thanks. LGTM. You can merge.

**Comment by rakeshm75** [comment]
One more point here, I can see that you have added `default_nettype none at the starting for the DSP modules (in cells_sim.v file). This will give compilation error as the wire definition is missing for all the inputs and outputs. You will need to add wire definition for all the inputs and outputs of all the modules below it (here...QL_DSP1, QL_DSP2, dsp_t1_sim, dsp_t1_20x18x64 & dsp_t1_10x9x32).

Ex: 
module QL_DSP1 (
    input [19:0] a,
    input [17:0] b,
    (* clkbuf_sink *)
    input clk0,
    (* clkbuf_sink *)
    input clk1,
    input [ 1:0] feedback0,
    input [ 1:0] feedback1,
    input load_acc0,
    input load_acc1,
    input reset0,
    input reset1,
    output reg [37:0] z
);
    parameter MODE_BITS = 27'b00000000000000000000000000;
endmodule  /* QL_DSP1 */

Should be changed to :

module QL_DSP1 (
    input wire [19:0] a,
    input wire [17:0] b,
    (* clkbuf_sink *)
    input wire clk0,
    (* clkbuf_sink *)
    input wire clk1,
    input wire [ 1:0] feedback0,
    input wire [ 1:0] feedback1,
    input wire load_acc0,
    input wire load_acc1,
    input wire reset0,
    input wire reset1,
    output reg [37:0] z
);
    parameter MODE_BITS = 27'b00000000000000000000000000;
endmodule  /* QL_DSP1 */

**Comment by lpawelcz** [comment]
@rakeshm75, right, I see that problem. I will fix this by moving `default_nettype none` to the top of the file and fixing up all the ports in all modules to make sure everything is ok. However, I would like to do that in a separate PR to keep things clean.
As for this PR, I will rebase it, solve conflicts and I belive it will be ok to merge then.

**Comment by lpawelcz** [comment]
@rakeshm75 I prepared PR for fixing the net type issue: https://github.com/chipsalliance/yosys-f4pga-plugins/pull/300
This PR is cleaned up and ready for merge.
